xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -130,64 +145,265 @@ bool CDVDSubtitlesLibass::DecodeHeader(char* data, int size)
+      style->BorderStyle = BORDER_STYLE_OUTLINE_BOX;
+      style->OutlineColour = outlineColor;
+      style->BackColour = ConvColor(subStyle.backgroundColor, subStyle.backgroundOpacity);
+      style->Outline = 1 * scale;
+      style->Shadow = 4; // Space between the text and the box edges
+    }
+
+    // Set the margins (in pixel)
+    style->MarginL = 20 * scale;
+    style->MarginR = style->MarginL;
+    style->MarginV = 20 * scale; // Vertical (direction depends on alignment)
+  }
+
+  // Set the alignment
+  if (subStyle.alignment == FontAlignment::TOP_LEFT)
+    style->Alignment = VALIGN_TOP | HALIGN_LEFT;
+  else if (subStyle.alignment == FontAlignment::TOP_CENTER)
+    style->Alignment = VALIGN_TOP | HALIGN_CENTER;
+  else if (subStyle.alignment == FontAlignment::TOP_RIGHT)
+    style->Alignment = VALIGN_TOP | HALIGN_RIGHT;
+  else if (subStyle.alignment == FontAlignment::MIDDLE_LEFT)
+    style->Alignment = VALIGN_CENTER | HALIGN_LEFT;
+  else if (subStyle.alignment == FontAlignment::MIDDLE_CENTER)
+    style->Alignment = VALIGN_CENTER | HALIGN_CENTER;
+  else if (subStyle.alignment == FontAlignment::MIDDLE_RIGHT)
+    style->Alignment = VALIGN_CENTER | HALIGN_RIGHT;
+  else if (subStyle.alignment == FontAlignment::SUB_LEFT)
+    style->Alignment = VALIGN_SUB | HALIGN_LEFT;
+  else if (subStyle.alignment == FontAlignment::SUB_CENTER)
+    style->Alignment = VALIGN_SUB | HALIGN_CENTER;
+  else if (subStyle.alignment == FontAlignment::SUB_RIGHT)
+    style->Alignment = VALIGN_SUB | HALIGN_RIGHT;
+
+  if (subtitleType == NATIVE)
+    ConfigureAssOverride(subStyle, style);
+}
+
+void CDVDSubtitlesLibass::ConfigureAssOverride(subtitlesStyle& subStyle, ASS_Style* style)
+{
+  // Default behaviour, disable ASS embedded styles override (if has been changed)
+  int stylesFlags = ASS_OVERRIDE_BIT_SELECTIVE_FONT_SCALE;
+
+  if (style)
+  {
+    // Manage override cases with ASS embedded styles
+    if (subStyle.assOverrideStyles == AssOverrideStyles::STYLES)
+    {
+      stylesFlags = ASS_OVERRIDE_BIT_FONT_SIZE_FIELDS | ASS_OVERRIDE_BIT_FONT_NAME |
+                    ASS_OVERRIDE_BIT_COLORS | ASS_OVERRIDE_BIT_ATTRIBUTES |
+                    ASS_OVERRIDE_BIT_BORDER | ASS_OVERRIDE_BIT_MARGINS;
+    }
+    else if (subStyle.assOverrideStyles == AssOverrideStyles::STYLES_POSITIONS)
+    {
+      stylesFlags = ASS_OVERRIDE_BIT_FONT_SIZE_FIELDS | ASS_OVERRIDE_BIT_FONT_NAME |
+                    ASS_OVERRIDE_BIT_COLORS | ASS_OVERRIDE_BIT_ATTRIBUTES |
+                    ASS_OVERRIDE_BIT_BORDER | ASS_OVERRIDE_BIT_MARGINS | ASS_OVERRIDE_BIT_ALIGNMENT;
+    }
+    else if (subStyle.assOverrideStyles == AssOverrideStyles::POSITIONS)
+    {
+      stylesFlags = ASS_OVERRIDE_BIT_ALIGNMENT;
+    }
+    ass_set_selective_style_override(m_renderer, style);
+  }
+  ass_set_selective_style_override_enabled(m_renderer, stylesFlags);
+}
+
+void CDVDSubtitlesLibass::ConfigureFont(bool overrideFont, std::string fontName)
+{
```

Comment:
A ternary operator would make this block of code more concise

Reply:
done